dgv/impact/task_functions.py

```diff
@@ -0,0 +1,188 @@
+from airflow.hooks.base import BaseHook
+from datagouvfr_data_pipelines.config import (
+    AIRFLOW_DAG_TMP,
+    AIRFLOW_DAG_HOME,
+    DATAGOUV_SECRET_API_KEY,
+    AIRFLOW_ENV,
+    MINIO_URL,
+    MINIO_BUCKET_DATA_PIPELINE_OPEN,
+    SECRET_MINIO_DATA_PIPELINE_USER,
+    SECRET_MINIO_DATA_PIPELINE_PASSWORD,
+)
+from datagouvfr_data_pipelines.utils.mattermost import send_message
+from datagouvfr_data_pipelines.utils.minio import send_files
+from datagouvfr_data_pipelines.utils.datagouv import (
+    get_all_from_api_query,
+    post_remote_resource,
+    DATAGOUV_URL
+)
+import pandas as pd
+import os
+from datetime import date, datetime
+from dateutil.relativedelta import relativedelta
+import numpy as np
+import requests
+import json
+
+TMP_FOLDER = f"{AIRFLOW_DAG_TMP}dgv_impact/"
+DATADIR = f"{TMP_FOLDER}data"
+
+
+def calculate_metrics():
+    # quality score
+    print("Calculating average quality score")
+    df_datasets = pd.read_csv(
+        # this is the catalog
+        "https://www.data.gouv.fr/fr/datasets/r/f868cca6-8da1-4369-a78d-47463f19a9a3",
+        dtype=str,
+        sep=";"
+    )
+    df_datasets["metric.views"] = df_datasets["metric.views"].astype(float)
+    df_datasets = df_datasets.sort_values(by="metric.views", ascending=False)
+    final = df_datasets[:1000]
+    final["quality_score"] = final["quality_score"].astype(float)
+    average_quality_score = round(100 * final["quality_score"].mean(), 2)
+
+    # time for legitimate answer
+    print("Calculating average time for legitimate answer")
+    datagouv_team = requests.get(
+        "https://www.data.gouv.fr/api/1/organizations/646b7187b50b2a93b1ae3d45/"
+    ).json()
+    datagouv_team = [m['user']['id'] for m in datagouv_team['members']]
+
+    r = get_all_from_api_query("https://www.data.gouv.fr/api/1/discussions/?sort=-created")
```

Comment:
You may use the discussion catalog: https://www.data.gouv.fr/fr/datasets/catalogue-des-donnees-de-data-gouv-fr/#/resources/d77705e1-4ecd-461c-8c24-662d47c4c2f9.
You won't have all infos, but since you do an additional call, I think it would be enough to filter on end and start date and messages length?

Reply:
As discussed, all the infos I need are not in this dataset so I will have to make API calls for each discussion anyway. I can use the dataset to filter on creation dates however, it changes the first part but not the rest of it. Could replace lines 53-65 with:
```
df_discussions = pd.read_csv('https://www.data.gouv.fr/fr/datasets/r/d77705e1-4ecd-461c-8c24-662d47c4c2f9', sep=';')
end_date = datetime.today().strftime("%Y-%m-%d")
oneyearago = date.today() - relativedelta(years=1)
start_date = oneyearago.strftime("%Y-%m-%d")
df_discussions = df_discussions.loc[df_discussions['created'].between(start_date, end_date)]
nb_discussions = 0
nb_discussions_with_legit_answer = 0
time_to_answer = []
k = 0
for discussion_id in df_discussions['id']:
    discussion = requests.get(
        f"https://www.data.gouv.fr/api/1/discussions/{discussion_id}/"
    ).json()
```
Thoughts?